src/nexusrpc/syncio/handler/_core.py

```diff
@@ -0,0 +1,293 @@
+from __future__ import annotations
+
+import concurrent.futures
+from typing import (
+    Any,
+    Callable,
+    Optional,
+    Sequence,
+    Union,
+    overload,
+)
+
+from typing_extensions import TypeGuard
+
+import nexusrpc
+from nexusrpc import InputT, OperationInfo, OutputT
+from nexusrpc._serializer import LazyValueT
+from nexusrpc._types import ServiceHandlerT
+from nexusrpc._util import (
+    get_callable_name,
+    is_async_callable,
+    set_operation_definition,
+    set_operation_factory,
+)
+from nexusrpc.handler._common import (
+    CancelOperationContext,
+    FetchOperationInfoContext,
+    FetchOperationResultContext,
+    StartOperationContext,
+    StartOperationResultAsync,
+    StartOperationResultSync,
+)
+from nexusrpc.handler._core import BaseServiceCollectionHandler
+from nexusrpc.handler._util import get_start_method_input_and_output_type_annotations
+
+from ...handler._operation_handler import OperationHandler
```

Comment:
Given that the OperationHandler works for both sync and async, this separate syncio module seems like overkill.

Reply:
What arrangement do you prefer? We have the following components that at the moment have sync and asynv versions:
- `Handler`
- `LazyValue`
- `sync_operation`

In any case, I think that what we should do is delete all of the above sync support, until we're sure how we want to present it. I'll do that in a follow-up PR.

To be clear for anyone else reading: when we "delete sync support", the SDK will continue to support user operation handlers with `def` methods (as long as an `executor` has been supplied).